Centaurus.Test.Domain/AlphaMessageHandlersTests.cs

```diff
@@ -0,0 +1,191 @@
+        const string auditorSecret = "SC4E2JXDOYKQYAZWKHRRNXYOJUYUJUPCU6LAWQ7JSJIV6AFHHW2M3LQU";
+
+
+        [SetUp]
+        public void Setup()
+        {
+            var settings = new AlphaSettings
+            {
+                HorizonUrl = "https://horizon-testnet.stellar.org",
+                NetworkPassphrase = "Test SDF Network ; September 2015",
+                Secret = KeyPair.Random().SecretSeed
+            };
+            settings.Build();
+
+            GlobalInitHelper.Setup(new string[] { clientSecret }, new string[] { auditorSecret }, settings);
+
+            MessageHandlers<AlphaWebSocketConnection>.Init();
+        }
+
+        [Test]
+        public async Task HandshakeTest()
+        {
+            Global.AppState.State = ApplicationState.Ready;
+
+            var clientKeyPair = KeyPair.FromSecretSeed(clientSecret);
+            var clientConnection = new AlphaWebSocketConnection(new FakeWebSocket());
+
+            var envelope = new HandshakeInit { HandshakeData = clientConnection.HandshakeData }.CreateEnvelope();
+            envelope.Sign(clientKeyPair);
+            var isHandled = await MessageHandlers<AlphaWebSocketConnection>.HandleMessage(clientConnection, envelope);
+
+            Assert.IsTrue(isHandled);
+        }
+
+        [Test]
+        public async Task HeartbeatTest()
+        {
+            Global.AppState.State = ApplicationState.Ready;
+
+            var clientKeyPair = KeyPair.FromSecretSeed(clientSecret);
+            var clientConnection = new AlphaWebSocketConnection(new FakeWebSocket());
+
+            var envelope = new Heartbeat().CreateEnvelope();
+            envelope.Sign(clientKeyPair);
+            var isHandled = await MessageHandlers<AlphaWebSocketConnection>.HandleMessage(clientConnection, envelope);
+
+            Assert.IsTrue(isHandled);
+        }
+
+        [Test]
+        [TestCase(clientSecret, ConnectionState.Validated, typeof(UnauthorizedException))]
+        [TestCase(auditorSecret, ConnectionState.Connected, typeof(InvalidStateException))]
+        [TestCase(auditorSecret, ConnectionState.Validated, null)]
+        public async Task SetApexCursorTest(string clientSecret, ConnectionState state, Type excpectedException)
+        {
+            try
+            {
+                Global.AppState.State = ApplicationState.Ready;
+
+                var clientKeyPair = KeyPair.FromSecretSeed(clientSecret);
+                var clientConnection = new FakeAlphaWebSocketConnection();
+                clientConnection.SetClientPubKey(clientKeyPair.PublicKey);
+                clientConnection.SetState(state);
+
+                var envelope = new SetApexCursor { Apex = 1 }.CreateEnvelope();
+                envelope.Sign(clientKeyPair);
+
+                var isHandled = await MessageHandlers<AlphaWebSocketConnection>.HandleMessage(clientConnection, envelope);
+
+                Assert.IsTrue(isHandled);
```

Comment:
Shouldn't we check the internal state change here? 
`HandleMessage` could always return `true` and tests will be all green despite the failed internal logic.